src/main/java/easy/mind/com/api/service/Impl/BeckDepressionInventoryServiceImpl.java

```diff
@@ -1,36 +1,48 @@
 package easy.mind.com.api.service.Impl;
 
+import easy.mind.com.api.DTO.BeckDepressionInventoryDTO;
+import easy.mind.com.api.DTO.UserDTO;
+import easy.mind.com.api.DTO.conversion.BeckDepressionToDTO;
 import easy.mind.com.api.entity.BeckDepressionInventory;
-import easy.mind.com.api.entity.User;
 import easy.mind.com.api.repository.BeckDepressionInventoryRepository;
 import easy.mind.com.api.service.BeckDepressionInventoryService;
-import org.springframework.beans.factory.annotation.Autowired;
+import easy.mind.com.api.service.UserService;
+import jakarta.persistence.EntityNotFoundException;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+
 import org.springframework.stereotype.Service;
 
+import java.util.List;
+
 @Service
+@RequiredArgsConstructor
 public class BeckDepressionInventoryServiceImpl implements BeckDepressionInventoryService {
 
-    private BeckDepressionInventoryRepository repository;
+    private final BeckDepressionInventoryRepository repository;
 
-    @Autowired
-    public BeckDepressionInventoryServiceImpl(BeckDepressionInventoryRepository repository) {
-        this.repository = repository;
-    }
+    private final UserService userService;
 
     @Override
-    public int getResult(BeckDepressionInventory inventory, User user) {
+    @Transactional
+    public BeckDepressionInventoryDTO create(int userId, BeckDepressionInventoryDTO inventoryDTO) {
 
+        BeckDepressionInventory inventory = BeckDepressionToDTO.convert(inventoryDTO);
+        UserDTO user = userService.readById(userId);
```

Comment:
you can replace it with `userService.throwIfNotExists(..)` here, and pass `userId` on the next line.
Also, this validation should be first. We first validating the input, and then proceed to business logic. It's more performant